### PR TITLE
[IMP] sale_project: hide type field in product form when creating SOL on the fly

### PR DESCRIPTION
**Pull request by kusi-web**
**Improvement**

When creating a SOL on the fly within a project/task, users can also create or edit products. To avoid errors from selecting non-service products, the type field is now hidden in the product form during this process, ensuring only service products are created.

task-4179188

### CODE REVIEW

File: addons/sale_project/views/sale_order_views.xml [hunk 1]
```diff
@@ -32,6 +32,17 @@
...
             </field>
+            <field name="product_template_id" position="attributes">
+                <attribute name="context">{
+                    'partner_id': parent.partner_id,
+                    'quantity': product_uom_qty,
+                    'pricelist': parent.pricelist_id,
+                    'uom':product_uom,
+                    'company_id': parent.company_id,
+                    'default_list_price': price_unit,
+                    'hide_product_type': context.get('hide_product_type'),
```
Comment (ppr-odoo) [inline addons/sale_project/views/sale_order_views.xml:43]: Why is this needed?

**Comment by ppr-odoo** [comment]
thanks for the work @kusi-odoo 

I do not prefer to hide the product type based on the context. If someone passes it, the product type will be hidden, and I think this approach is risky.

I propose a second option: We should extend the product form view in primary mode and hide the product type there. Additionally, we need to pass the reference of this product view in the field context.

**Comment by kusi-odoo** [comment]
Thanks for the suggestion @ppr-odoo 
I've updated the code as per your suggestion. Please review it when you have time. 
Thanks!

File: addons/sale_project/views/sale_order_line_views.xml [hunk 2]
```diff
@@ -28,6 +28,7 @@
                 <attribute name="context">{
                     'default_type': 'service',
                     'default_service_policy': 'ordered_prepaid',
+                    'form_view_ref': 'sale_project.product_template_form_view_inherit_sale_project'
```
Comment (ppr-odoo) [inline addons/sale_project/views/sale_order_line_views.xml:31]:
```suggestion
                    'form_view_ref': 'sale_project.product_template_form_view_inherit_sale_project',
```

**Review by MissingNoShiny — APPROVED** [review]
LGTM, thanks! :slightly_smiling_face:

**Review by xavierbol — COMMENTED** [review]
LGTM, thanks for your work! 🙂 

robodoo r+

**Comment by xavierbol** [comment]
I just rebase your branch. 🙂 

robodoo r+
